### ISSUE TITLE
dyld: Symbol not found: _SecTrustEvaluateWithError  when upgrade to go 1.18 in macOS High Sierra 10.13.3

### ISSUE THREAD

**Issue by raygecao** [golang/go#52757]
### What version of Go are you using (`go version`)?

`1.18.1` 

### Does this issue reproduce with the latest release?

Yes

### What did you do?

I upgraded go from go 1.17.2 (worked well) to go1.18.1 (from `go1.18.1.darwin-amd64.pkg` in the official download page), then a horrible error was thrown.

My system was `MacOS High Sierra 10.13.3`, and I find the it's supported in the release note of go 1.17 (while go1.18 doesn't point the requirement of darwin). 

It's same to https://github.com/golang/go/issues/50855. However my system met the requirement.

The problem is resolved when I upgrade my darwin to `MacOS Mojave 10.14.6`

### What did you expect to see?

go version go1.18.1 darwin/amd64

### What did you see instead?

```shell
$ /usr/local/go/bin/go version

dyld: Symbol not found: _SecTrustEvaluateWithError
  Referenced from: /usr/local/go/bin/go
  Expected in: flat namespace

[1]    67235 abort      /usr/local/go/bin/go
```

**Comment by seankhliao** [comment]
previously #52112

**Comment by dr2chase** [comment]
This looks like a duplicate, and perhaps solved merely by updating 10.13 to 10.13.6 or better.
